java/src/main/native/src/TableJni.cpp

```diff
@@ -935,6 +935,39 @@ jlongArray combine_join_results(JNIEnv *env, cudf::table &left_results,
   return combine_join_results(env, std::move(left_cols), std::move(right_cols));
 }
 
+cudf::column_view remove_validity_from_col(cudf::column_view column_view) {
+  if (column_view.nullable() && column_view.null_count() == 0) {
+    // null_mask is allocated but no nulls present therefore we create a new column_view without the
+    // null_mask to avoid things blowing up in reading the parquet file
+    if (!cudf::is_nested(column_view.type())) {
+      cudf::column_view cv(column_view.type(), column_view.size(), column_view.data<int32_t>(),
+                           nullptr, 0);
+      return cv;
+    } else {
+      std::vector<cudf::column_view> children;
+      children.reserve(column_view.num_children());
+      for (auto it = column_view.child_begin(); it != column_view.child_end(); it++) {
+        children.push_back(remove_validity_from_col(*it));
+      }
+      cudf::column_view cv(column_view.type(), column_view.size(), column_view.data<int32_t>(),
+                           nullptr, 0, 0, children);
+      return cv;
+    }
+  } else {
+    return column_view;
+  }
+}
+
+cudf::table_view remove_validity_if_needed(cudf::table_view *input_table_view) {
```

Comment:
As a side note, it might be good to try and put this into the C++ part of cudf as a stand alone API.  From your tests Spark has trouble reading maps from parquet where the key is nullable even if there are no null values. That means anyone trying to write out maps in parquet through cudf is going to be producing files that are not compatible with Spark. If these are official cudf APIs, then they have the option to fix it., even if it is just for the keys to maps.

Reply:
Since each call to `null_count()` could call a kernel, ideally this would be more efficient as a single-kernel across all bitmasks in a table: Reference #8365

Reply:
That would also be a great API to have.  One that can look at all of the columns in a table recursively, see which already have `null_count`s cached, and then calculate/cache them for any that are missing.

Reply:
In interest of time, I would love to do #8365 as a follow-on if you are OK with it.

Reply:
I am fine with it as a follow on.